lib/AST/DeclContext.cpp

```diff
@@ -880,6 +880,24 @@ DeclRange IterableDeclContext::getMembers() const {
   return DeclRange(FirstDeclAndLazyMembers.getPointer(), nullptr);
 }
 
+DeclRange IterableDeclContext::getMembersThatMightBeTypes() const {
+  switch (getIterableContextKind()) {
+  case IterableDeclContextKind::NominalTypeDecl: {
+    auto nominal = cast<NominalTypeDecl>(this);
+    if (nominal->hasClangNode()) {
+      return DeclRange(DeclIterator(), DeclIterator());
```

Comment:
These optimizations aren't correct for C structs, Objective-C classes, or extensions thereof, which *can* have type members due to import-as-member. It's only Objective-C protocols (and extensions thereof) that are guaranteed not to have nested types.

Reply:
Do we really care about all nested types here, or just associated types of protocols? Maybe it would be good to push the whole thing down into ProtocolDecl.

Reply:
*Technically,* import-as-member always makes a new extension that doesn't have a Clang node. But enums and structs can have nested types (RawValue, for example).

Reply:
Oh, I don't know what that is! But I will narrow scope accordingly :)